toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -363,11 +377,22 @@ func (r *RpmRepoCloner) CloneRawPackageNames(cloneDeps bool, rawPackageNames ...
 
 	logger.Log.Debugf("Will clone in total %d items.", len(rawPackageNames))
 
+	// Create a list of lists for each transaction. Each transaction will be cloned separately. Generally either all
+	// packages will be cloned in a single transaction or each package will be cloned in its own transaction.
+	transactions := [][]string{}
+	if singleTransaction {
+		transactions = append(transactions, rawPackageNames)
+	} else {
+		for _, packageName := range rawPackageNames {
+			transactions = append(transactions, []string{packageName})
+		}
+	}
+
 	allPackagesPrebuilt = true
-	for _, packageNameToClone := range rawPackageNames {
-		logger.Log.Debugf("Cloning raw name (%s).", packageNameToClone)
+	for _, packageNamesToClone := range transactions {
+		logger.Log.Debugf("Cloning raw names (%v).", packageNamesToClone)
```

Comment:
Since we always trickle down to this function, I'm curious if we could replace "raw names" with a simple "packages". The "raw" part may not be super clear to everyone.